front/app/utils/dateUtils.ts

```diff
@@ -249,3 +249,46 @@ export function calculateRoundedEndDate(
   endDate.setMinutes(startDate.getMinutes() + durationInMinutes);
   return endDate;
 }
+
+export const userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+// Why do we need this function?
+// The backend sends dates in the format "YYYY-MM-DD" without a time component.
+// When we parse this date in the frontend, it is interpreted as
+// midnight in UTC.
+// This means that if we are west of UTC, e.g. in Brazil,
+// The date will be interpreted as 21:00 the previous day.
+// This function makes sure that the date is always interpreted as midnight in the user's timezone.
+const backendDatestringRegex = /^\d{4}-\d{2}-\d{2}$/;
+
+export const parseBackendDateString = (dateString?: string) => {
```

Comment:
I think I finally solved all the stupid 'one day off' errors. We just have to make sure to use this utility whenever we receive a string from the backend that has the format `yyyy-MM-DD`